falcon_kit/mains/rr_ctg_track.py

```diff
@@ -0,0 +1,167 @@
+    inputs = []
+    for fn in file_list:
+        if len(fn) != 0:
+            inputs.append( (run_tr_stage1, db_fn, fn, min_len, bestn, rid_to_ctg) )
+    """
+    Aggregate hits from each individual LAS and keep the best n hit. 
+    Note that this does not guarantee that the final results is globally the best n hits espcially 
+    when the number of `bestn` is too small.  In those case, if there is more hits from single LAS
+    file, then we will miss some good  hits.
+    """
+    bread_to_areads = {}
+    for fn, res in exe_pool.imap(io.run_func, inputs):
+        for k in res:
+            bread_to_areads.setdefault(k, [])
+            for item in res[k]:
+                if len(bread_to_areads[k]) < bestn:
+                    heappush( bread_to_areads[k], item )
+                else:
+                    heappushpop( bread_to_areads[k], item )
+
+    #rid_to_oid = open(os.path.join( rawread_dir, "raw_read_ids" ) ).read().split("\n")
+
+    """
+    For each b-read, we find the best contig map throgh the b->a->contig map.
+    """
+    with open( os.path.join(asm_dir, "read_maps/rawread_to_contigs"), "w") as out_f:
+        for bread in bread_to_areads:
+
+            ctg_score = {}
+            for s, rid in bread_to_areads[bread]:
+                if rid not in rid_to_ctg: continue
+                
+                ctgs = rid_to_ctg[rid]
+                for ctg in ctgs:
+                    ctg_score.setdefault(ctg, [0,0])
+                    ctg_score[ctg][0] += -s
+                    ctg_score[ctg][1] += 1
+
+            #oid = rid_to_oid[int(bread)]
+            ctg_score = ctg_score.items()
+            ctg_score.sort( key = lambda k: k[1][0] )
+            rank = 0
+
+            for ctg, score_count in ctg_score:
+                if bread in rid_to_ctg and ctg in rid_to_ctg[bread]:
+                    in_ctg = 1
+                else:
+                    in_ctg = 0
+                score, count = score_count
+                #print(bread, oid, ctg, count, rank, score, in_ctg, file=out_f) 
+                print(bread, ctg, count, rank, score, in_ctg, file=out_f) 
+                rank += 1
+
+        
+
+def try_run_track_reads(n_core, base_dir, min_len, bestn):
+    io.LOG('starting track_reads')
+    
+    rawread_dir = os.path.abspath( os.path.join(base_dir, "0-rawreads") )
+
+    # better logic for finding the las files path or move the logic to extern (taking the --fofn option?) 
+    file_list = glob.glob( os.path.join( rawread_dir, "m*/raw_reads.*.las") )
+    io.LOG('file list: %r' % file_list)
+
+    # same, shoud we decide this as a parameter
+    db_fn = os.path.join( rawread_dir, "raw_reads.db" )
+    n_core = min(n_core, len(file_list))
+    exe_pool = Pool(n_core)
+    try:
+        run_track_reads(exe_pool, base_dir, file_list, min_len, bestn, db_fn)
+        io.LOG('finished track_reads')
+    except KeyboardInterrupt:
+        io.LOG('terminating track_reads workers...')
+        exe_pool.terminate()
```

Comment:
And re-raise. Otherwise, you've suppressed the KeyboardInterrupt.

Also, this can be a bare `except:` because we want to terminate on *any* exception.